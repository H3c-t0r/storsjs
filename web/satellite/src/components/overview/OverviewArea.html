<!--Copyright (C) 2019 Storj Labs, Inc.-->
<!--See LICENSE for copying information.-->

<div class="overview-area">
    <div class="overview-area__welcome-container">
        <h1 class="overview-area__welcome-container__title">Welcome to Storj</h1>
        <p class="overview-area__welcome-container__sub-title">
            To create a project, get started by adding a payment method - <br/>your first 5GB are free for your first project.
        </p>
        <router-link class="overview-area__welcome-container__button" :to="billingPath" >Add Payment</router-link>
    </div>
    <div class="overview-area__steps-area">
        <div class="overview-area__steps-area__numbers">
            <FirstStepIcon class="overview-area__steps-area__numbers__icon"/>
            <div class="overview-area-divider"></div>
            <SecondStepIcon class="overview-area__steps-area__numbers__icon"/>
            <div class="overview-area-divider"></div>
            <ThirdStepIcon class="overview-area__steps-area__numbers__icon"/>
        </div>
        <div class="overview-area__steps-area__items">
            <div class="overview-area__steps-area__items__add-payment-info">
                <h2 class="overview-area__steps-area__items__add-payment-info__title">Add Payment Info</h2>
                <img class="overview-area-image" src="@/../static/images/overview/payment.jpg" alt="payment image">
            </div>
            <div class="overview-area__steps-area__items__create-project">
                <h2 class="overview-area__steps-area__items__create-project__title">Create a Project</h2>
                <img class="overview-area-image" src="@/../static/images/overview/project.jpg" alt="project image">
            </div>
            <div class="overview-area__steps-area__items__create-api-key">
                <h2 class="overview-area__steps-area__items__create-api-key__title">Create an API Key</h2>
                <img class="overview-area-image" src="@/../static/images/overview/apikey.jpg" alt="api key image">
            </div>
        </div>
    </div>
</div>
