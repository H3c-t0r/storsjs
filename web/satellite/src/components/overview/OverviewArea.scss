// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

p,
h1 {
    margin: 0;
}

.overview-area {
    width: auto;
    height: calc(100% - 150px);
    padding: 75px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    font-family: 'font_regular', sans-serif;

    &__welcome-container {
        width: auto;
        background-color: #fff;
        border-radius: 6px;
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: space-between;
        padding: 73px 261px;
        margin-bottom: 32px;

        &__title {
            font-family: 'font_bold', sans-serif;
            font-size: 32px;
            line-height: 39px;
            color: #1b2533;
            margin-bottom: 20px;
        }

        &__sub-title {
            font-size: 16px;
            line-height: 24px;
            color: #354049;
            margin-bottom: 30px;
            text-align: center;
        }

        &__button {
            font-size: 16px;
            line-height: 23px;
            padding: 10px 15px;
            color: #fff;
            background-color: #2683ff;
            border-radius: 8px;

            &:hover {
                background-color: #0059d0;
            }
        }
    }

    &__steps-area {
        padding: 55px;
        width: auto;
        background-color: #fff;
        border-radius: 12px;

        &__numbers {
            display: flex;
            flex-direction: row;
            align-items: center;
            justify-content: space-between;
            padding: 0 125px;
            margin-bottom: 15px;

            &__icon {
                min-width: 30px;
                min-height: 30px;
            }
        }

        &__items {
            display: flex;
            justify-content: space-between;
            align-items: center;

            &__add-payment-info,
            &__create-project,
            &__create-api-key {
                display: flex;
                flex-direction: column;
                align-items: center;
                justify-content: space-between;

                &__title {
                    font-family: 'font_bold', sans-serif;
                    font-size: 24px;
                    line-height: 16px;
                    text-align: center;
                    color: #354049;
                }
            }

            &__add-payment-info,
            &__create-project {
                margin-right: 20px;
            }
        }
    }
}

.overview-area-divider {
    width: calc(46% + 2px);
    border: 1px solid #cfd5da;
}

@media screen and (max-width: 1450px) {

    .overview-area {

        &__welcome-container {
            width: 626px;
            padding: 73px 55px;
        }

        &__steps-area {

            &__numbers {
                padding: 0 80px;
            }

            &__items {

                &__add-payment-info,
                &__create-project,
                &__create-api-key {

                    &__title {
                        font-family: 'font_bold', sans-serif;
                        font-size: 19px;
                        line-height: 16px;
                        text-align: center;
                        color: #354049;
                    }
                }
            }
        }
    }

    .overview-area-image {
        max-width: 210px;
        max-height: 150px;
    }
}
