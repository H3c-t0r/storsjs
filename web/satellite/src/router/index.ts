import Vue from 'vue';
import Router from 'vue-router';
import ROUTES from '@/utils/constants/routerConstants';
import Login from '@/views/Login.vue';
import Register from '@/views/Register.vue';

Vue.use(Router);

export default new Router({
	mode: 'history',
	routes: [
		{
			path: ROUTES.DEFAULT.path,
			name: ROUTES.DEFAULT.name,
			component: Login
		},
		{
			path: ROUTES.REGISTER.path,
			name: ROUTES.REGISTER.name,
			component: Register
		}
  	]
});
