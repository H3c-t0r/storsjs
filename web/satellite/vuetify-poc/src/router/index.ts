// Copyright (C) 2023 Storj Labs, Inc.
// See LICENSE for copying information.

// Composables
import { createRouter, createWebHistory } from 'vue-router'

const routes = [
  {
    path: '/vuetifypoc',
    redirect: { path: '/projects' }, // redirect
    component: () => import('@poc/layouts/default/Default.vue'),
    children: [
      {
        path: '/dashboard',
        name: 'Dashboard',
        component: () => import(/* webpackChunkName: "home" */ '@poc/views/Dashboard.vue'),
      },
      {
        path: '/buckets',
        name: 'Buckets',
        component: () => import(/* webpackChunkName: "Buckets" */ '@poc/views/Buckets.vue'),
      },
      {
        path: '/bucket',
        name: 'Bucket',
        component: () => import(/* webpackChunkName: "Buckets" */ '@poc/views/Bucket.vue'),
      },
      {
        path: '/access',
        name: 'Access',
        component: () => import(/* webpackChunkName: "Access" */ '@poc/views/Access.vue'),
      },
      {
        path: '/team',
        name: 'Team',
        component: () => import(/* webpackChunkName: "Team" */ '@poc/views/Team.vue'),
      },
    ],
  },
  {
    path: '/account',
    component: () => import('@poc/layouts/default/Account.vue'),
    children: [
      {
        path: '/billing',
        name: 'Billing',
        component: () => import(/* webpackChunkName: "Billing" */ '@poc/views/Billing.vue'),
      },
      {
        path: '/account-settings',
        name: 'Account Settings',
        component: () => import(/* webpackChunkName: "MyAccount" */ '@poc/views/AccountSettings.vue'),
      },
      {
        path: '/design-library',
        name: 'Design Library',
        component: () => import(/* webpackChunkName: "DesignLibrary" */ '@poc/views/DesignLibrary.vue'),
      },
    ],
  },
  {
    path: '/projects',
    component: () => import('@poc/layouts/default/AllProjects.vue'),
    children: [
      {
        path: '/projects',
        name: 'Projects',
        component: () => import(/* webpackChunkName: "Projects" */ '@poc/views/Projects.vue'),
      },
    ],
  },
]

const router = createRouter({
  history: createWebHistory(),
  routes,
})

export default router
