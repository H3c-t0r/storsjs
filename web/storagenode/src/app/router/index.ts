// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import Vue from 'vue';
import Router from 'vue-router';

import { NavigationLink } from '@/app/types/navigation';
import DashboardArea from '@/app/views/DashboardArea.vue';
import NotificationsArea from '@/app/views/NotificationsArea.vue';

Vue.use(Router);

export abstract class RouteConfig {
    public static Root = new NavigationLink('', 'Root');
    public static Notifications = new NavigationLink('/notifications', 'Notifications');
}

const router = new Router({
    mode: 'history',
    routes: [
        {
            path: RouteConfig.Root.path,
            name: RouteConfig.Root.name,
            component: DashboardArea
        },
        {
            path: RouteConfig.Notifications.path,
            name: RouteConfig.Notifications.name,
            component: NotificationsArea
        },
    ]
});

export default router;
