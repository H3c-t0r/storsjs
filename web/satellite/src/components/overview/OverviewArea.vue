// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template src="./OverviewArea.html"></template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import VButton from '@/components/common/VButton.vue';

import FirstStepIcon from '@/../static/images/common/one.svg';
import ThirdStepIcon from '@/../static/images/common/three.svg';
import SecondStepIcon from '@/../static/images/common/two.svg';

import { RouteConfig } from '@/router';

@Component({
    components: {
        VButton,
        FirstStepIcon,
        ThirdStepIcon,
        SecondStepIcon,
    },
})
export default class OverviewArea extends Vue {
    public readonly billingPath: string = RouteConfig.Account.with(RouteConfig.Billing).path;
}
</script>

<style scoped lang="scss" src="./OverviewArea.scss"></style>
