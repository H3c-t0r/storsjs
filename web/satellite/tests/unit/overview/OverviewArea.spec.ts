// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import OverviewArea from '@/components/overview/OverviewArea.vue';

import { router } from '@/router';
import { mount } from '@vue/test-utils';

describe('OverviewArea.vue', () => {
    it('renders correctly', (): void => {
        const wrapper = mount(OverviewArea, {
            router,
        });

        expect(wrapper).toMatchSnapshot();
    });
});
