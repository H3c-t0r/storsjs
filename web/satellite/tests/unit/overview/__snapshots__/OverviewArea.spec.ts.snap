// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewArea.vue renders correctly 1`] = `
<div class="overview-area">
  <div class="overview-area__welcome-container">
    <h1 class="overview-area__welcome-container__title">Welcome to Storj</h1>
    <p class="overview-area__welcome-container__sub-title">
      To create a project, get started by adding a payment method - <br>your first 5GB are free for your first project.
    </p> <a href="/account/billing" class="overview-area__welcome-container__button">Add Payment</a>
  </div>
  <div class="overview-area__steps-area">
    <div class="overview-area__steps-area__numbers"><svg width="30" height="30" viewBox="0 0 30 30" fill="none" xmlns="http://www.w3.org/2000/svg" class="overview-area__steps-area__numbers__icon">
        <circle cx="15" cy="15" r="15" fill="#519CFF"></circle>
        <path d="M17.0916 9.36364H14.7791L11.8984 11.1875V13.3693L14.5632 11.6989H14.6314V21H17.0916V9.36364Z" fill="white"></path>
      </svg>
      <div class="overview-area-divider"></div> <svg width="30" height="30" viewBox="0 0 30 30" fill="none" xmlns="http://www.w3.org/2000/svg" class="overview-area__steps-area__numbers__icon">
        <circle cx="15" cy="15" r="15" fill="#519CFF"></circle>
        <path d="M11.4041 21H19.6996V18.9886H14.8132V18.9091L16.5121 17.2443C18.9041 15.0625 19.5462 13.9716 19.5462 12.6477C19.5462 10.6307 17.8984 9.20455 15.4041 9.20455C12.9609 9.20455 11.2848 10.6648 11.2905 12.9489H13.6257C13.62 11.8352 14.3246 11.1534 15.3871 11.1534C16.4098 11.1534 17.1712 11.7898 17.1712 12.8125C17.1712 13.7386 16.603 14.375 15.5462 15.392L11.4041 19.2273V21Z" fill="white"></path>
      </svg>
      <div class="overview-area-divider"></div> <svg width="30" height="30" viewBox="0 0 30 30" fill="none" xmlns="http://www.w3.org/2000/svg" class="overview-area__steps-area__numbers__icon">
        <circle cx="15" cy="15" r="15" fill="#519CFF"></circle>
        <path d="M15.4709 21.1591C18.0845 21.1591 19.9538 19.7216 19.9482 17.733C19.9538 16.2841 19.0334 15.25 17.3232 15.0341V14.9432C18.6243 14.7102 19.522 13.7898 19.5163 12.483C19.522 10.6477 17.9141 9.20455 15.505 9.20455C13.1186 9.20455 11.3232 10.6023 11.2891 12.6136H13.647C13.6754 11.7273 14.4879 11.1534 15.4936 11.1534C16.4879 11.1534 17.1527 11.7557 17.147 12.6307C17.1527 13.5455 16.3743 14.1648 15.255 14.1648H14.1697V15.9716H15.255C16.5732 15.9716 17.397 16.6307 17.3913 17.5682C17.397 18.4943 16.6016 19.1307 15.4766 19.1307C14.3913 19.1307 13.5788 18.5625 13.5334 17.7102H11.0561C11.0959 19.7443 12.9141 21.1591 15.4709 21.1591Z" fill="white"></path>
      </svg>
    </div>
    <div class="overview-area__steps-area__items">
      <div class="overview-area__steps-area__items__add-payment-info">
        <h2 class="overview-area__steps-area__items__add-payment-info__title">Add Payment Info</h2> <img src="@/../static/images/overview/payment.jpg" alt="payment image" class="overview-area-image">
      </div>
      <div class="overview-area__steps-area__items__create-project">
        <h2 class="overview-area__steps-area__items__create-project__title">Create a Project</h2> <img src="@/../static/images/overview/project.jpg" alt="project image" class="overview-area-image">
      </div>
      <div class="overview-area__steps-area__items__create-api-key">
        <h2 class="overview-area__steps-area__items__create-api-key__title">Create an API Key</h2> <img src="@/../static/images/overview/apikey.jpg" alt="api key image" class="overview-area-image">
      </div>
    </div>
  </div>
</div>
`;
